File: packages/wazeko-types/src/group.ts
import { Jid } from "./jid.js";

export type ParticipantRole = "member" | "admin" | "superadmin";

export interface GroupParticipant {
  id: Jid;
  role: ParticipantRole;
}

export interface GroupMetadata {
  id: Jid;
  subject: string;
  description?: string;
  owner?: Jid;
  creationTime: number;
  participants: GroupParticipant[];
  restrict: boolean;
  announce: boolean;
}

File: crates/wazeko-types/src/group.rs
use crate::jid::Jid;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
pub enum ParticipantRole {
    Member,
    Admin,
    SuperAdmin,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct GroupParticipant {
    pub id: Jid,
    pub role: ParticipantRole,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct GroupMetadata {
    pub id: Jid,
    pub subject: String,
    pub description: Option<String>,
    pub owner: Option<Jid>,
    pub creation_time: u64,
    pub participants: Vec<GroupParticipant>,
    pub restrict: bool,
    pub announce: bool,
}
